### PR TITLE
test: parametrized engine contract test suite

### LINKED ISSUE #139: test: Parametrized engine contract test suite
## Summary

Each backend has its own integration test file (~400-535 lines each):

- `test/integration/test_jax_backend.py` (535 lines)
- `test/integration/test_jax_md_backend.py` (436 lines)
- `test/integration/test_openmm_backend.py`
- `test/integration/test_tinker_backend.py`

These files are largely copy-paste with the same test structure (energy at equilibrium, energy displaced, gradients, frequencies, Hessian, cross-engine parity) but different engine construction. `test_backend_optimizer_matrix.py` parametrizes across backends but is the exception.

## Proposed Changes

1. Create a single parametrized contract test suite:

```python
@pytest.fixture(params=available_test_engines())
def engine(request):
    return get_engine(request.param)

class TestMMEngineContract:
    def test_energy_at_equilibrium(self, engine, h2_molecule, h2_ff):
        assert engine.energy(h2_molecule, h2_ff) == pytest.approx(0.0, abs=1e-6)

    def test_gradient_matches_fd(self, engine, displaced_molecule, ff):
        ...

    def test_frequencies_sorted(self, engine, molecule, ff):
        ...
```

2. Engine-specific tests (JIT compilation, neighbor lists) stay in separate files
3. **Contract tests** run on every engine automatically -- adding a new engine gets all contract tests for free

## Benefits

- Adding JAX-MD took 436 lines of tests. With parametrized contract tests, it would be ~50 lines (engine factory + edge cases)
- Guarantees all engines meet the same behavioral contract
- Reduces test maintenance burden significantly

## Files Involved

- `test/integration/test_engine_contract.py` (new -- parametrized suite)
- `test/integration/test_jax_backend.py` -- keep engine-specific tests only
- `test/integration/test_jax_md_backend.py` -- keep engine-specific tests only
- `test/integration/test_openmm_backend.py` -- keep engine-specific tests only
- `test/integration/test_tinker_backend.py` -- keep engine-specific tests only

### PR DESCRIPTION
**Pull request by ericchansen**
## Summary

Adds a parametrized contract test suite that verifies every available MM engine satisfies the `MMEngine` ABC contract. Uses the engine registry from #134 to discover engines at collection time — new engines automatically get tested when they register.

## What's tested

13 contract tests × N available engines (currently 3: OpenMM, JAX, JAX-MD):

| Category | Tests |
|----------|-------|
| **Metadata** | `name` returns non-empty string, `is_available()` is True, `supported_functional_forms()` non-empty |
| **Energy** | Returns float, near zero at equilibrium, increases when displaced |
| **Hessian** | Shape is (3N, 3N), symmetric |
| **Frequencies** | Returns list/array, count equals 3N |
| **Minimize** | Returns tuple with 3+ elements, lowers energy, water molecule (medium tier) |

## Duplicate removal

Removed 6 tests from individual backend files that were exact duplicates of contract behavior:
- `test_jax_backend.py`: `test_is_available`, `test_hessian_shape`, `test_hessian_symmetric`, `test_frequencies_count`
- `test_jax_md_backend.py`: `test_is_available`, `test_hessian_shape`, `test_hessian_symmetric`

Kept all engine-specific extended tests (near-zero translation modes, analytical gradients, cross-engine parity, optimizer integration).

## Other changes

- `mm/__init__.py`: Added `JaxMDEngine` export, simplified try/except pattern (removed `_HAS_JAX_ENGINE` flag)

## Test results

467 passed, 86 skipped (36 new contract tests, 6 duplicates removed → net +30)

Closes #139

### CODE REVIEW

File: q2mm/backends/mm/__init__.py [hunk 1]
```diff
@@ -2,19 +2,31 @@
...
 
-__all__ = ["OpenMMEngine", "TinkerEngine"]
-if _HAS_JAX_ENGINE:
-    __all__ = [*__all__, "JaxEngine"]
+try:
+    from q2mm.backends.mm.jax_md_engine import JaxMDEngine
+
+    __all__ = [*__all__, "JaxMDEngine"]
+except ImportError:
+    pass
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/backends/mm/__init__.py]: Fixed in 67923e9 — same change, both engines now imported unconditionally.
